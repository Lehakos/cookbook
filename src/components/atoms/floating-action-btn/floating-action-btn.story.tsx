import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';

import { FloatingActionBtn } from './floating-action-btn';

const stories = storiesOf('Atoms|FloatingActionBtn', module);

stories.add('Default', () => (
  <FloatingActionBtn
    ariaLabel="plus"
    fixedPosition={boolean('fixedPosition', false)}
  >
    +
  </FloatingActionBtn>
));
