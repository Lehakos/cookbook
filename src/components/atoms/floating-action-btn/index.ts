export { FloatingActionBtn } from './floating-action-btn';
