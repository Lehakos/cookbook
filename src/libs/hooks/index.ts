export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
