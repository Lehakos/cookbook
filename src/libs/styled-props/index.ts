export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
