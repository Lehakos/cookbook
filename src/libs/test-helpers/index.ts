export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
