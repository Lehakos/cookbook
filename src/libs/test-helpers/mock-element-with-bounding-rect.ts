export const mockElementWithBoundingRect = ({
  width = 100,
  height = 100,
  top = 0,
  left = 0,
}) => {
  const right = left + width;
  const bottom = top + height;

  Object.defineProperties(Element.prototype, {
    offsetLeft: {
      get() {
        return left;
      },
      configurable: true,
    },
    offsetTop: {
      get() {
        return top;
      },
      configurable: true,
    },
    offsetHeight: {
      get() {
        return height;
      },
      configurable: true,
    },
    offsetWidth: {
      get() {
        return width;
      },
      configurable: true,
    },
  });

  Element.prototype.getBoundingClientRect = jest.fn(() => {
    return {
      width,
      height,
      top,
      left,
      bottom,
      right,
    };
  });

  const element = document.createElement('div');

  return element;
};
