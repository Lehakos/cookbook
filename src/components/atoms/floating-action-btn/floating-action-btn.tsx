import React from 'react';

import * as s from './floating-action-btn.styles';

interface Props {
  disabled?: boolean;
  /**
   * Description for screenreaders
   */
  ariaLabel: string;
  /**
   * Show button in right bottom angle with position: fixed
   * @default true
   */
  fixedPosition?: boolean;
}

export const FloatingActionBtn: React.FunctionComponent<Props> = function({
  children,
  disabled,
  ariaLabel,
  fixedPosition = true,
  ...rest
}) {
  return (
    <s.Wrapper
      {...rest}
      aria-label={ariaLabel}
      fixedPosition={fixedPosition}
      disabled={disabled}
    >
      {children}
    </s.Wrapper>
  );
};
