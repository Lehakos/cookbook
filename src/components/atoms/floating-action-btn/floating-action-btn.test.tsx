import React from 'react';

import { render } from 'libs/test-helpers';

import { FloatingActionBtn } from './floating-action-btn';

describe('<FloatingActionBtn />', () => {
  it('renders properly', () => {
    const { container } = render(
      <FloatingActionBtn ariaLabel="Plus">+</FloatingActionBtn>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
