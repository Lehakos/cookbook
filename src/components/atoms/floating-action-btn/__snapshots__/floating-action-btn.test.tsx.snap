// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatingActionBtn /> renders properly 1`] = `
.c0 {
  height: 64px;
  width: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  padding: 0;
  border: none;
  color: #fff;
  background-color: #F47983;
  border-radius: 50%;
  box-shadow: 0 0 4px 0 rgba(0,0,0,.2);
  cursor: pointer;
  position: fixed;
  bottom: 24px;
  right: 24px;
  z-index: 100;
}

.c0:hover {
  background-color: #f3707a;
}

.c0:active {
  background-color: #f2616d;
}

.c0[disabled] {
  background-color: #f79aa2;
  cursor: default;
}

<button
  aria-label="Plus"
  class="c0"
>
  +
</button>
`;
